internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,327 @@
+
+			id := customlocations.NewCustomLocationID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{
+				ClusterExtensionIds: pointer.To(model.ClusterExtensionIds),
+				DisplayName:         pointer.To(model.DisplayName),
+				HostResourceId:      pointer.To(model.HostResourceId),
+				HostType:            pointer.To(customlocations.HostType(model.HostType)),
+				Namespace:           pointer.To(model.Namespace),
+			}
+
+			if len(model.Authentication) > 0 {
+				auth := model.Authentication[0]
+				customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+					Type:  pointer.To(auth.Type),
+					Value: pointer.To(auth.Value),
+				}
+			}
+
+			props := customlocations.CustomLocation{
+				Id:         pointer.To(id.ID()),
+				Location:   model.Location,
+				Name:       pointer.To(model.Name),
+				Properties: pointer.To(customLocationProps),
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, props); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:              id.CustomLocationName,
+					ResourceGroupName: id.ResourceGroupName,
+					Location:          model.Location,
+				}
+
+				if props.ClusterExtensionIds != nil {
+					state.ClusterExtensionIds = pointer.From(props.ClusterExtensionIds)
+				}
+				if props.DisplayName != nil {
+					state.DisplayName = pointer.From(props.DisplayName)
+				}
+				if props.HostResourceId != nil {
+					state.HostResourceId = pointer.From(props.HostResourceId)
+				}
+				if props.HostType != nil {
+					state.HostType = string(pointer.From(props.HostType))
+				}
+				if props.Namespace != nil {
+					state.Namespace = pointer.From(props.Namespace)
+				}
+
+				if props.Authentication != nil && props.Authentication.Type != nil && props.Authentication.Value != nil {
```

Comment:
As above, pointer.From will deal with nils, so this can become
```suggestion
				if props.Authentication != nil {
```

Reply:
I've opened an issue to them and leave the link in the comment line.